contrib/babelfishpg_tsql/src/procedures.c

```
@@ -2062,3 +2067,180 @@ gen_sp_droprolemember_subcmds(const char *user, const char *member)
+		CommandCounterIncrement();
+
+		// parsetree_list = gen_sp_rename_subcmds(obj_name, new_name, schema_name, objtype);
+		parsetree_list = gen_sp_rename_subcmds(obj_name, new_name, schema_name, objtype_code);
+
+		//5. run all commands
+		foreach(parsetree_item, parsetree_list)
+        {
+			Node *stmt = ((RawStmt *) lfirst(parsetree_item))->stmt;
+			PlannedStmt *wrapper;
+			
+			/* need to make a wrapper PlannedStmt */
+			wrapper = makeNode(PlannedStmt);
+			wrapper->commandType = CMD_UTILITY;
+			wrapper->canSetTag = false;
+			wrapper->utilityStmt = stmt;
+			wrapper->stmt_location = 0;
+			wrapper->stmt_len = 16;
+
+			/* do this step */
+			ProcessUtility(wrapper,
+				pstrdup(process_util_querystr),
+				// "(ALTER TABLE )",
+				false,
+				PROCESS_UTILITY_SUBCOMMAND,
+				NULL,
+				NULL,
+				None_Receiver,
+				NULL);
+
+			/* make sure later steps can see the object created here */
+			CommandCounterIncrement();
+        }
+	}
+	PG_CATCH();
+	{
+		set_config_option("babelfishpg_tsql.sql_dialect", saved_dialect,
+							(superuser() ? PGC_SUSET : PGC_USERSET),
+							PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);
+		PG_RE_THROW();
+	}
+	PG_END_TRY();
+	set_config_option("babelfishpg_tsql.sql_dialect", saved_dialect,
+							(superuser() ? PGC_SUSET : PGC_USERSET),
+							PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);
+	PG_RETURN_VOID();
+}
+
+static List *
+gen_sp_rename_subcmds(const char *objname, const char *newname, const char *schemaname, ObjectType objtype)
+{
+	StringInfoData query;
+	List *res;
+	Node *stmt;
+	RenameStmt *renamestmt;
+
+	initStringInfo(&query);
+	if (objtype == OBJECT_TABLE) {
+		appendStringInfo(&query, "ALTER TABLE dummy RENAME TO dummy; ");
+	} else if (objtype == OBJECT_VIEW) {
+		appendStringInfo(&query, "ALTER VIEW dummy RENAME TO dummy; ");
+	} else if (objtype == OBJECT_PROCEDURE) {
+		appendStringInfo(&query, "ALTER PROCEDURE dummy RENAME TO dummy; ");
+	// } else if (objtype == OBJECT_FUNCTION) {
+	} else {
```

Need to throw an error if subtype is not supported